jumpscale/sals/reservation_chatflow/solutions.py

```diff
@@ -148,11 +147,53 @@ def list_kubernetes_solutions(self, next_action=NextAction.DEPLOY, sync=True):
                         "Slave Pools": [],
                         "Master Pool": workload.info.pool_id,
                     }
-                    result[name].update(self.get_workload_capacity(workload))
                     if workload.master_ips:
                         result[name]["Slave IPs"].append(workload.ipaddress)
         return list(result.values())
 
+    def get_kubernetes_solution_details(self, k8s_name, next_action=NextAction.DEPLOY, sync=True):
+        if sync:
+            j.sals.reservation_chatflow.deployer.load_user_workloads(next_action=next_action)
+        if not sync and not j.sals.reservation_chatflow.deployer.workloads[next_action][WorkloadType.Kubernetes]:
+            j.sals.reservation_chatflow.deployer.load_user_workloads(next_action=next_action)
+        results = []
+        for kube_workloads in j.sals.reservation_chatflow.deployer.workloads[next_action][
+            WorkloadType.Kubernetes
+        ].values():
+
+            for workload in kube_workloads:
+                if not workload.info.metadata:
+                    continue
+                try:
+                    metadata = j.data.serializers.json.loads(workload.info.metadata)
+                except:
+                    continue
+                if not metadata.get("form_info"):
+                    continue
+                name = metadata["form_info"].get("Solution name", metadata.get("name"))
+                if name == k8s_name:
+                    node = {}
+                    role = "master"
+                    # Slave Node
+                    if workload.master_ips:
+                        role = "slave"
+
+                    # Get public ip
+                    public_ip = ""
+                    if workload.public_ip:
+                        workload_public_ip = j.sals.zos.get().workloads.get(workload.public_ip)
+                        public_ip = workload_public_ip.ipaddress.split("/")[0] if workload_public_ip else ""
+
+                    node = {"role": role, "wid": workload.id, "ip_address": workload.ipaddress, "public_ip": public_ip}
+                    # Handle Storage object
+                    disk_size = self.get_workload_capacity(workload).get("Disk Size")
+                    node.update(self.get_workload_capacity(workload))
```

Comment:
why calling the same method twice? you can call it once and pop whatever values you need